### DISCUSSION TITLE
Wasm for indexing

### DISCUSSION THREAD

**Discussion by SbiCA** [ravendb/ravendb#16135] [Ideas]
Not sure if there is anything to the thought but how about using WASI to run indexes on raven. 

Should that improve perf if you're able to compile JS/TS to WASM and run it with something like https://github.com/SteveSandersonMS/DotNetIsolator instead of Jint or V8?

https://youtu.be/gxplZcW5t3Y 
https://www.wasm.builders/gunjan_0307/compiling-javascript-to-wasm-34lk

**Comment by ayende** [comment]
That is a really interesting option, and certainly something that we'll explore.

Right now we are held up on the 6.0 release, but we'll look at that in more details afterward, thanks.

**Reply by ppekrol** [reply]
https://issues.hibernatingrhinos.com/issue/RavenDB-20227/Research-WASI-and-check-if-we-can-use-it-for-JavaScript-projections-and-Indexing
